support-lib/jni/djinni_support.cpp

```diff
@@ -239,8 +239,13 @@ jint JniEnum::ordinal(JNIEnv * env, jobject obj) const {
 
 LocalRef<jobject> JniEnum::create(JNIEnv * env, jint value) const {
     LocalRef<jobject> values(env, env->CallStaticObjectMethod(m_clazz.get(), m_staticmethValues));
+    jniExceptionCheck(env);
     DJINNI_ASSERT(values, env);
-    return LocalRef<jobject>(env, env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()), value));
+    LocalRef<jobject> result(env,
+                             env->GetObjectArrayElement(static_cast<jobjectArray>(values.get()),
+                                                        value));
+    jniExceptionCheck(env);
+    return std::move(result);
```

Comment:
Good to know.  I would've assumed RVO would take precedence there too, since the compiler knows the const is meaningless, but I see it's a case where the standard specifies behavior in terms of a return "of the same type" so const or rvalue-ref kills it.  That's tricker to get right than I'd have hoped.  And is a reason not to make variables const, which makes me sad.